buildAndReleaseTask/prComment.ts

```diff
@@ -0,0 +1,156 @@
+import {readFileSync} from "fs";
+import {join as pathJoin} from "path";
+
+import * as azdev from "azure-devops-node-api";
+
+import * as tl from "azure-pipelines-task-lib/task";
+
+import { IGitApi } from "azure-devops-node-api/GitApi";
+import { Comment, CommentThread, CommentThreadStatus, CommentType } from "azure-devops-node-api/interfaces/GitInterfaces";
+
+export const PULUMI_LOG_FILENAME = "pulumi-out.log";
+
+const SOURCE_PULUMI = "Pulumi";
+const defaultCommentThreadProperties = {
+    "source": SOURCE_PULUMI
+};
+
+function getFormattedPulumiLogs(pulumiLogs: string, pulCommand: string, pulStackFqdn: string): string {
+    const heading = `#### :tropical_drink: \`${pulCommand}\` on ${pulStackFqdn}`;
+    const summary = '<summary>Pulumi report</summary>';
+    const body = `
+${heading}
+<details>
+${summary}
+
+\`\`\`
+${pulumiLogs}
+\`\`\`
+
+</details>
+`;
+
+    return body;
+}
+
+function getLastCommentInThread(thread: CommentThread): Comment | undefined {
+    if (!thread.comments?.length) {
+        return;
+    }
+
+    const numComments = thread.comments.length;
+    return thread.comments[numComments - 1];
+}
+
+/**
+ * Creates a PR comment thread if the current build is a PR build.
+ */
+export async function createPrComment() {
```

Comment:
Would you be able to refactor this function to use dependency injection? Specifically for the task inputs.
I'd prefer if locations where `tl.getBool`, `tl.getInput` etc. were be collected into a struct within the calling function and passed into this function as a parameter. That would allow us (in the future) to unit test `createPrComment` (mocking out `createThread` and `createComment` or other side-effect carrying functions).

Reply:
@RobbieMcKinstry I've refactored this function to the extent that the task inputs can be passed in as args. However, I kind of think we should keep the lookup of build env vars local to this function rather than pulling it up. The reason I say that is, the way it is right now is kind of nice that the code that executes the pulumi command and calls this function does not need to know which env vars it needs to read and how it should handle validating them. That's the job of `createPrComment`. I did, however, introduce some types to help with passing around the build env info. Please have a look at it and let me know what you think.

Reply:
Looks good, thank you Praneet! I appreciate it :)